content/tutorials/publish-subscribe.textile

```diff
@@ -589,14 +604,49 @@ blang[android].
   "See this step in Github":https://github.com/ably/tutorials/commit/pubsub-android-step3
 
 blang[javascript].
+  First, create a variable to store a reference to the channel:
+
   ```[javascript]
-    var channel = realtime.channels.get("sport");
-    channel.subscribe(function(msg) {
-      alert("Received: " + JSON.stringify(msg.data));
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY";
+    const realtime = new Ably.Realtime(apiKey);
+    let channel = null;
+  </script>
+  ```
+
+  Then, ensure that you are connected to the Ably platform, by monitoring the @connected@ event:
+
+  ```[javascript]
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY";
+    const realtime = new Ably.Realtime(apiKey);
+    let channel = null;
+    realtime.connection.on("connected", () => {
+      console.log("Connected to Ably!");
+    });
+  </script>
+  ```
+
+  Within the @connected@ event handler, use @realtime.channels.get@ to get a reference to the "sport" channel and then subscribe to the channel:
+
+  ```[javascript]
+  <script>
+    const apiKey = "YOUR_ABLY_API_KEY";
+    const realtime = new Ably.Realtime(apiKey);
+    let channel = null;
+    realtime.connection.on("connected", () => {
```

Comment:
Was the code meant to be updated to use the promises interface, or was this a quick fix, with promises to come later?

Reply:
@tbedford Initially it was supposed to be a quick fix, but I've spoken to Phil about this and I've updated the code to use the promises interface. So it will probably need looking at again ... sorry!